build/int.cloudbuild.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+timeout: 14400s
+steps:
+- id: prepare
+  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+  args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && prepare_environment']
+  env:
+  - 'TF_VAR_org_id=$_ORG_ID'
+  - 'TF_VAR_folder_id=$_FOLDER_ID'
+  - 'TF_VAR_billing_account=$_BILLING_ACCOUNT'
+- id: test-all-samples
```

Comment:
Does splitting tasks at least for apply and destroy make sense? I think it provides better readability when view logs in cloud build.

Reply:
The reason for not splitting up is because some resources have the same name in samples and I wanted to minimize the changes to the samples themselves. For example sample A may create resource with name A and if we continue with all applies first, there can a conflict when sample B tries to create same resource with name A. 

In the current approach since we destroy before proceeding, this will not happen. I do have some followups for log readability including using https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/1234